vscode-dotnet-runtime-library/src/Acquisition/AcquisitionInvoker.ts

```diff
@@ -237,19 +238,26 @@ At dotnet-install.ps1:1189 char:5
         let error = null;
         let command = null;
 
-        const possibleCommands =
+        const possibleCommands = // create a bunch of commands that just return the index of the correct shell in powershell
             [
-                CommandExecutor.makeCommand(`powershell.exe`, []),
-                CommandExecutor.makeCommand(`%SystemRoot%\\System32\\WindowsPowerShell\\v1.0\\powershell.exe`, []),
-                CommandExecutor.makeCommand(`pwsh`, []),
-                CommandExecutor.makeCommand(`powershell`, []),
-                CommandExecutor.makeCommand(`pwsh.exe`, [])
+                CommandExecutor.makeCommand('0', []),
+                CommandExecutor.makeCommand('1', []),
+                CommandExecutor.makeCommand('2', []),
+                CommandExecutor.makeCommand('3', []),
+                CommandExecutor.makeCommand('4', []),
             ];
-
+        const possiblePowershellPaths =
+            [ // use shell as powershell and see if it passes or not. This is faster than doing it with the default shell, as that spawns a cmd to spawn a pwsh
+                { shell: `%SystemRoot%\\System32\\WindowsPowerShell\\v1.0\\powershell.exe` },

```

Comment:
These variables can change. You really should be calling `SHGetSpecialFolderPathW`. Does TS not support any native APIs at all?

Q2: What is the significance of the ordering here?

Reply:
Why check for `pwsh.exe` and `pwsh`? Isn't one for Windows and the other for Mac? Should you construct this based on the OS instead?

Reply:
Q2: The ordering determines which one it checks first, and if it works then it uses that one.
![image](https://github.com/user-attachments/assets/b221d177-644c-4c20-8a19-a40fc37e4ae8)

I think your API suggestion is a good idea and the better way to go but https://github.com/search?q=SHGetSpecialFolderPathW+language%3Atypescript&type=code that doesn't seem to ever be done before. I think I will investigate this but don't want to block this PR for it.

Why check for pwsh.exe and pwsh and powershell.exe and powershell? Based on the data there is a very odd number of people where it works for one but not the other. For the (Vast) majority the system root one works.
![image](https://github.com/user-attachments/assets/b4c5f8f5-8259-44cd-84ac-b72c52418c5e)

Reply:
sounds good. Agree not to block on the special folder change, that should probably be looked at on its own,

Reply:
Based on my sample data literally nobody in this large set: didnt have poweshell.exe but did have pwsh.exe, so I can remove that one. But there were a small number who had powershell but not powershell.exe. Pwsh is the newer version of pwsh. This is all Windows only so why is this the case, I dont know ._.